src/nfa/operations.cc

```diff
@@ -151,11 +151,11 @@ namespace {
                                     const State Tid, const StateSet& T,                      // current state to check
                                     Nfa& result) {
 
-        std::unordered_map<StateSet, State>::iterator it = subset_map.begin();
+        auto it = subset_map.begin();
```

Comment:
Why use `auto` here? Isn't it better to see the type explicitly?

It is iterator though, so maybe it makes sense.

Reply:
For iterators, the general best practice is to use `auto` and do not explicitly state types of the iterators. This allows you to switch the underlying data structures without having to modify larger portions of the code.

Reply:
True, makes sense